store/sqlstore/reaction_store.go

```diff
@@ -167,23 +178,80 @@ func (s *SqlReactionStore) PermanentDeleteBatch(endTime int64, limit int64) (int
 	return rowsAffected, nil
 }
 
-func saveReactionAndUpdatePost(transaction *gorp.Transaction, reaction *model.Reaction) error {
-	if err := transaction.Insert(reaction); err != nil {
-		return err
+func (s *SqlReactionStore) saveReactionAndUpdatePost(transaction *gorp.Transaction, reaction *model.Reaction) error {
+	params := map[string]interface{}{
+		"UserId":    reaction.UserId,
+		"PostId":    reaction.PostId,
+		"EmojiName": reaction.EmojiName,
+		"CreateAt":  reaction.CreateAt,
+		"UpdateAt":  reaction.UpdateAt,
 	}
 
+	if s.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		if _, err := transaction.Exec(
+			`INSERT INTO
+				Reactions
+				(UserId, PostId, EmojiName, CreateAt, UpdateAt, DeleteAt)
+			VALUES
+				(:UserId, :PostId, :EmojiName, :CreateAt, :UpdateAt, 0)
+			ON DUPLICATE KEY UPDATE
+				UpdateAt = :UpdateAt, DeleteAt = 0`, params); err != nil {
+			return err
+		}
+	} else if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
```

Comment:
With the last min-requirements update on postgres, can we now use the "ON CONFLICT" to solve this? (Probably out of the scope of the PR, but maybe is interesting to create a ticket for that).

Reply:
Yes, there are three areas we do upserts that would benefit from ON CONFLICT.  If our minimum is at 9.5 or better I'll create a ticket for those.

Reply:
Our minimum is 10.0 now.

Reply:
Ticket to use ON CONFLICT  https://mattermost.atlassian.net/browse/MM-31906

Reply:
Now using ON CONFLICT https://github.com/mattermost/mattermost-server/pull/16654/commits/474429d278b1475ed9b7ec1690b3611b291b0360